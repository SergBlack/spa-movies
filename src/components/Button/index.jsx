import React, { useContext } from 'react';
import { bool, number, string } from 'prop-types';
import styled, { ThemeContext } from 'styled-components';

const StyledButton = styled.button`
  width: ${({ width }) => width};
  height: ${({ height }) => height};
  opacity: ${({ opacity }) => opacity};
  background-color: ${({ bgColor }) => bgColor};
  color: ${({ textColor }) => textColor};
  font-size: 20px;
  text-transform: uppercase;
  border-radius: 5px;
  border: none;
  padding: 10px;
  margin: 10px;

  :hover {
    background-color: ${({ hoverColor }) => hoverColor};
    cursor: pointer;
  }

  :disabled {
    cursor: not-allowed;
    opacity: 0.8;
  }
`;

const Button = ({
  text,
  height,
  width,
  color,
  opacity,
  disabled,
}) => {
  const { mainColors, textColorsDependBgColor, shadesMainColors } = useContext(ThemeContext);

  return (
    <StyledButton
      height={height}
      width={width}
      bgColor={mainColors[color]}
      textColor={textColorsDependBgColor[color]}
      hoverColor={shadesMainColors[color]}
      opacity={opacity}
      disabled={disabled}
    >
      {text}
    </StyledButton>
  );
};

Button.propTypes = {
  text: string.isRequired,
  height: string,
  width: string,
  color: string,
  opacity: number,
  disabled: bool,
};

Button.defaultProps = {
  height: '60px',
  width: '200px',
  color: 'red',
  opacity: 1,
  disabled: false,
};

export default Button;
