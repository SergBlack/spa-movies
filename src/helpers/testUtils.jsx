/* eslint-disable react/prop-types */
import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import { theme } from '../App';
import rootReducer from '../redux/rootReducer';

function render(
  ui,
  {
    initialState,
    store = createStore(rootReducer, initialState),
    ...renderOptions
  } = {},
) {
  function Wrapper({ children }) {
    return (
      <ThemeProvider theme={theme}>
        <Provider store={store}>
          {children}
        </Provider>
      </ThemeProvider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';
export { render };
