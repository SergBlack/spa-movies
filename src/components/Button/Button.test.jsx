import React from 'react';
import { render } from '@helpers/testUtils';

import Button from './index';

describe('Button', () => {
  test('renders Button snapshot', () => {
    const { asFragment } = render(
      <Button
        type="button"
        text="Text button"
        height="60px"
        width="200px"
        color="red"
        shape="square"
        uppercase
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
