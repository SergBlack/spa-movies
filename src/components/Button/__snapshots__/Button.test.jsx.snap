// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders Button snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-iBPRYJ eyuqg"
    height="60px"
    opacity="1"
    shape="square"
    type="button"
    width="200px"
  >
    Text button
  </button>
</DocumentFragment>
`;
